contracts/FIFOMarket.sol

```diff
@@ -27,18 +32,23 @@ contract FIFOMarket is
   IERC777RecipientUpgradeable
 {
   using RemovalUtils for uint256;
+  using EnumerableSetUpgradeable for EnumerableSetUpgradeable.UintSet;
 
   IERC1820RegistryUpgradeable private _erc1820;
   Removal private _removal;
   Certificate private _certificate;
   BridgedPolygonNORI private _bridgedPolygonNori;
-  mapping(uint256 => uint256) private _queue;
-  uint256 private _queueHeadIndex;
-  uint256 private _queueNextInsertIndex;
   address private _noriFeeWallet;
   uint256 private _noriFee;
   uint256 public priorityRestrictedThreshold;
-  uint256 public totalSupply;
+  uint256 public totalNumberActiveRemovals;
+  uint256 public totalActiveSupply;
+  uint256 public totalReservedSupply;
+  uint256 public activeSupplierCount;
+  address private _currentSupplierAddress;
+  mapping(address => RoundRobinOrder) private _suppliersInRoundRobinOrder;
+  EnumerableSetUpgradeable.UintSet private _reservedSupply;
```

Comment:
could a [bitmap](https://hiddentao.com/archives/2018/12/10/using-bitmaps-for-efficient-solidity-smart-contracts) be a better solution for this?

[OZ has a util for bitmaps here](https://docs.openzeppelin.com/contracts/4.x/api/utils#BitMaps) 

I would suggest this happens in a separate PR but I'd like to hear your thoughts or at least a spike ticket created

Reply:
Thanks for the suggestion, this was from Amie's original PR and I was very much planning on changes in a follow-up PR. Will look into this approach.